File: difference_of_psd.jl
function difference_of_psd(A)
if ishermitian(A)
    S = eigen(A);
    D = S.values;
    V = S.vectors;
    P = (D+abs.(D))/2;
    B = V*Diagonal(P)*V'; B = (B+B')/2;
    C = B-A;
    return(B,C)
end
end

File: difference_of_psd.m
function [B,C] = difference_of_psd(A)
if ishermitian(A)
    [V,D] = eig(A);
    D = sparse(D);
    P = (D+abs(D))/2;
    B = V*P*V'; B = (B+B')/2;
    C = B-A;
end
end